src/core/thread/osthread.d

```diff
@@ -3479,47 +3438,42 @@ extern (C) @nogc nothrow
 
 version (LDC)
 {
-    package(core.thread) void* getStackTop() nothrow @nogc @naked
+    version (X86)      version = LDC_stackTopAsm;
+    version (X86_64)   version = LDC_stackTopAsm;
+    version (ARM_Any)  version = LDC_stackTopAsm;
+    version (PPC_Any)  version = LDC_stackTopAsm;
+    version (MIPS_Any) version = LDC_stackTopAsm;
+
+    version (LDC_stackTopAsm)
     {
-        /* The inline assembler is written in a style that the code can be
-         * inlined.
-         * The use of intrinsic llvm_frameaddress is a reasonable default for
-         * cpu architectures without assembler support from LLVM. Because of
-         * the slightly different meaning the code must not be inlined.
+        /* The inline assembler is written in a style that the code can be inlined.
+         * If it isn't, the function is still naked, so the caller's stack pointer
+         * is used nevertheless.
          */
-        version (X86)
-        {
-            return __asm!(void*)("movl %esp, $0", "=r");
-        }
-        else version (X86_64)
-        {
-            return __asm!(void*)("movq %rsp, $0", "=r");
-        }
-        else version (AArch64)
-        {
-            return __asm!(void*)("mov $0, sp", "=r");
-        }
-        else version (ARM)
-        {
-            return __asm!(void*)("mov $0, sp", "=r");
-        }
-        else version (PPC)
-        {
-            return __asm!(void*)("mr $0, 1", "=r");
-        }
-        else version (PPC64)
-        {
-            return __asm!(void*)("mr $0, 1", "=r");
-        }
-        else version (MIPS32)
-        {
-            return __asm!(void*)(".set noat; move $0, $$sp; .set at", "=r");
-        }
-        else version (MIPS64)
+        package(core.thread) void* getStackTop() nothrow @nogc @naked
         {
-            return __asm!(void*)("move $0, $$sp", "=r");
+            version (X86)
+                return __asm!(void*)("movl %esp, $0", "=r");
+            else version (X86_64)
+                return __asm!(void*)("movq %rsp, $0", "=r");
+            else version (ARM_Any)
+                return __asm!(void*)("mov $0, sp", "=r");
+            else version (PPC_Any)
+                return __asm!(void*)("mr $0, 1", "=r");
+            else version (MIPS_Any)
+                return __asm!(void*)("move $0, $$sp", "=r");
+            else
+                static assert(0);
         }
-        else
+    }
+    else
+    {
+        /* The use of intrinsic llvm_frameaddress is a reasonable default for
+         * cpu architectures without assembler support from LLVM. Because of
+         * the slightly different meaning the function must neither be inlined
+         * nor naked.
+         */
+        package(core.thread) void* getStackTop() nothrow @nogc
```

Comment:
This was only refactored because I think the previously *naked* `getStackToo()` would have returned a wrong stack-top (caller's frame base) in this fallback case.